### ISSUE TITLE
Add unit tests

### ISSUE THREAD

**Issue by rubensworks** [joachimvh/asyncjoin#9]
I just noticed that this repo doesn't have any unit tests.
Might be good to add some, given the critical nature of this project :-)

**Comment by joachimvh** [comment]
I completely agree! But I don't think I will be able to make time for this any time soon so PRs are always accepted 😉
